packages/omgidl-parser/src/processIDL.ts

```diff
@@ -26,15 +26,22 @@ export function buildMap(definitions: AnyASTNode[]): Map<string, AnyIDLNode> {
       const newNode = makeIDLNode(scopePath, node, idlMap);
       idlMap.set(newNode.scopedIdentifier, newNode);
       if (node.declarator === "enum") {
-        let nextImplicitIEnumValue = 0;
-        const enumConstants = node.enumerators.map((m) => ({
-          declarator: "const" as const,
-          isConstant: true as const,
-          name: m.name,
-          type: "unsigned long",
-          value: getValueAnnotation(m.annotations) ?? (nextImplicitIEnumValue++ as ConstantValue),
-          isComplex: false,
-        }));
+        // How C++ does implicit enums is that they will increment after the last explicit enum
+        // even if it collides with an existing enum
+        // initialize to -1 so that first value is 0
+        let prevEnumValue = -1;
+        const enumConstants = node.enumerators.map((m) => {
+          const enumValue = getValueAnnotation(m.annotations) ?? (++prevEnumValue as ConstantValue);
+          prevEnumValue = enumValue as number;
```

Comment:
why do we need `as number`? isn't it already a number? could this be solved by putting `: number` on line 32?